### PR TITLE
wip: ccloudvm: Add templates for kata containers.

### PR DESCRIPTION
**Pull request by jcvenegas**
Add templates to tests packages in VM.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

File: ccloudvm/kata-pkgs-fedora27.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+base_image_url: https://mirror.us-midwest-1.nexcess.net/fedora/releases/27/CloudImages/x86_64/images/Fedora-Cloud-Base-27-1.6.x86_64.qcow2
```
Comment (jodh-intel) [inline ccloudvm/kata-pkgs-fedora27.yaml]: This URL doesn't look that "official" wrt the Fedora project - could you maybe find a fedoraproject one instead?

**Comment by jodh-intel** [comment]
Hi @markdryan - could you take a look at this?

**Comment by markdryan** [comment]
You could simplify both workloads a fair bit by inheriting from the basic xenial and fedora workloads in the ccloudvm/workloads directory.

For xenial we even have a docker-xenial workload that installs the latest version of docker and correctly configures proxies if defined.

This is how the ciao workload works, e.g.,

https://github.com/intel/ccloudvm/blob/master/workloads/ciao.yaml#L2

Note that it inherits from docker-xenial and so does not need to install docker, configure a user account or setup apt proxies.

**Comment by jcvenegas** [comment]
@jodh-intel  @markdryan  I like the proposed ideas - I will move this files to ccloudvm to ensure it inherits  from the files there.

**Comment by markdryan** [comment]
@jcvenegas There's no need to move the files to ccloudvm.  I fact I think it's probably better to keep them in the kata-containers project.  You should still be able to inherit from the ccloudvm workloads, even if your workload ins't itself in ccloudvm.

**Comment by jcvenegas** [comment]
@markdryan  - got it .

**Comment by jcvenegas** [comment]
@markdryan  @jodh-intel  
- Added only docker-xenial based workload. 
- Added a README.

**Comment by markdryan** [comment]
@jcvenegas Have you tested behind a proxy?

I wonder 

wget -qO - http://download.opensuse.org/repositories/home:/katacontainers:/release/xUbuntu_$(lsb_release -rs)/Release.key | sudo apt-key add -

will work behind a proxy. Could you check?

**Comment by jcvenegas** [comment]
@markdryan  I am using redsocks proxy in my host :/

**Comment by markdryan** [comment]
We should make sure that the workload works in a proxied environment without having to use a transparent proxy.

**Comment by jcvenegas** [comment]
@markdryan by inheriting xenial -> docker-xenial -> kata-docker-xenial.yaml   I want to think that will use the proxy setup from https://github.com/intel/ccloudvm/blob/master/workloads/xenial.yaml#L9-L18

**Comment by jcvenegas** [comment]
@markdryan  sure - let me disble my redsocks and export proxy manually

**Comment by jcvenegas** [comment]
that task failed ! :(

File: ccloudvm/kata-docker-xenial.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (grahamwhaley) [inline ccloudvm/kata-docker-xenial.yaml]: typo in commit subject:
s/worklaod/workload/

Comment (grahamwhaley) [reply ccloudvm/kata-docker-xenial.yaml]: and also in the body:
s/craete/create/

File: ccloudvm/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Test Kata using ccloudvm
+
+The tool [ccloudvm](https://github.com/intel/ccloudvm/) is  command line
```
Comment (grahamwhaley) [inline ccloudvm/README.md]: double space? '  ' ? Maybe it should say 'is a command'...

File: ccloudvm/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+tool to create development and demo environments. The tool sets up these development
+environments inside a virtual machine.
+
+## How to use this Kata workloads for `ccloudvm`
+
+- Follow the `ccloudvm` [install instructions](https://github.com/intel/ccloudvm/#introduction)
+
+- Run a Kata Containers workloads.
```
Comment (grahamwhaley) [inline ccloudvm/README.md]: I don't understand what you mean here - this line feels maybe redundant?

File: ccloudvm/README.md (hunk 3 above)
Comment (jodh-intel) [inline ccloudvm/README.md]: I'd switch the word order a little:

> The [ccloudvm](https://github.com/intel/ccloudvm/) tool is a command line

File: ccloudvm/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+### Create Docker\* and Kata Containers virtualized enviroment
+
+Create a virtual machine with Docker and Kata containers.
+
+```bash
+$ ccloudvm create --name YOUR_VM_NAME $PWD/kata-docker-xenial.yaml
+$ ccloudvm connect YOUR_VM_NAME
+```
+
+You are ready to use kata with docker in a virtualized enviroment.
```
Comment (jodh-intel) [inline ccloudvm/README.md]: s/kata/Kata/

File: ccloudvm/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Test Kata using ccloudvm
```
Comment (jodh-intel) [inline ccloudvm/README.md]: Bonus points for adding a table of contents :smile:

File: ccloudvm/README.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+### Create Docker\* and Kata Containers virtualized enviroment
+
+Create a virtual machine with Docker and Kata containers.
+
+```bash
+$ ccloudvm create --name YOUR_VM_NAME $PWD/kata-docker-xenial.yaml
+$ ccloudvm connect YOUR_VM_NAME
+```
+
+You are ready to use Kata with docker in a virtualized enviroment.
```
Comment (amshinde) [inline ccloudvm/README.md]: s/enviroment/environment/

File: ccloudvm/README.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+to create development and demo environments. The tool sets up these development
+environments inside a virtual machine.
+
+## How to use Kata workloads for `ccloudvm`
+
+- Follow the `ccloudvm` [install instructions](https://github.com/intel/ccloudvm/#introduction)
+
+### Create Docker\* and Kata Containers virtualized enviroment
+
+Create a virtual machine with Docker and Kata containers.
```
Comment (amshinde) [inline ccloudvm/README.md]: s/containers/Containers/

Comment (amshinde) [reply ccloudvm/README.md]: I think this should this be "Create a virtual machine with Docker and Kata Containers installed." ?

**Pull request merged**